src/index.js

```diff
@@ -65,53 +66,67 @@ module.exports = {
     // If we are on Netlify pull the secrets that have been added through the UI
     // so that we can contextualize them as well.
     if (isNetlify) {
-      console.log('Adding Netlify Build secrets to secrets array');
       secrets = { ...secrets, ...netlifyConfig.build.environment };
+
+      // Ensure that dotenv gets pushed to the external bundle.
+      if (!Array.isArray(netlifyConfig.functions['*'].external_node_modules)) {
+        netlifyConfig.functions['*'].external_node_modules = [];
+      }
+      netlifyConfig.functions['*'].external_node_modules.push('dotenv');
+
+      // Ensure that the .env file we are writing gets bundled with the func.
+      if (!Array.isArray(netlifyConfig.functions['*'].included_files)) {
+        netlifyConfig.functions['*'].included_files = [];
+      }
+      netlifyConfig.functions['*'].included_files.push('.env');
```

Comment:
This line ensures that the `.env` we write during the build process gets added to the function bundle when it is zipped-and-shipped.